### PR TITLE
Import prod script

### PR DESCRIPTION
**Pull request by EGCGDanForth**
3 new scripts

1) update_freezer_location - populates submitted sample UDFs from a step UDF
2)next_step_assignment - automatically assigns the next step of all samples in a step as either the next step in the protocol or, if it is the last step, then assigns it as "Mark Protocol as Complete"
3)next_step_assignment-review - automatically assigns the next step of all samples as escalate to manager review

Updated prepare_discard_plate so includes"valid" plate name suffixes rather than excludes "invalid" suffixes.

### CODE REVIEW

File: scripts/next_step_assignment.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+        protocol = Protocol(self.process.lims, uri='/'.join(self.process.step.configuration.uri.split('/')[:-2]))
+
+        actions = self.process.step.actions  # obtain the actions of the step # creates a StepActions entity for the current step
+
+        next_actions = actions.next_actions  # obtain the next actions in the step #creates a list of dict for next_actions for the step
+
+        steps = protocol.steps  # a list of all the ProtocolSteps in protocol
+
+        # print(steps)
```
Comment (katieemelianova) [inline scripts/next_step_assignment.py]: remove print statement

File: scripts/next_step_assignment.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+        for step in steps:  # find the index of the current step in the list of all ProtocolSteps
+            if step == current_step:
```
Comment (katieemelianova) [inline scripts/next_step_assignment.py]: Are you able here to just get the last step (steps[-1]) and see if that == current step?

File: scripts/next_step_assignment.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                current_step_index = steps.index(step)
+
+        # if the current step index plus 1 matches the number of steps in the protocol (i.e. length of the steps list) then the current step must be the last step in
+        # the protocol so the next action should be complete. If the current step index plus 1 is less than the length then next action should be next step
+
+        if current_step_index + 1 == len(steps):  # where index values run 0 to X and length values run 1 to X
+            # print("current_step_index +1 == len(steps)")
```
Comment (katieemelianova) [inline scripts/next_step_assignment.py]: remove print statement

**Comment by katieemelianova** [comment]
Perhaps remove some of the excess whitespace between lines within functions.

File: scripts/update_freezer_location.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+class UpdateFreezerLocation(StepEPP):
+    def _run(self):
+
+        samples = self.samples
+
+        for sample in samples:
+            sample.udf['Freezer'] = self.process.udf.get('New Freezer Location')
+            sample.udf['Shelf'] = self.process.udf.get('New Freezer Location')
+            sample.udf['Box'] = self.process.udf.get('New Freezer Location')
+            sample.put()
```
Comment (mwhamgenomics) [inline scripts/update_freezer_location.py]: Do we really want to set all 3 UDFs to exactly the same value?

File: scripts/next_step_assignment.py (hunk 2 above)
Comment (katieemelianova) [inline scripts/next_step_assignment.py]: I'm not sure you necessarily have to loop through steps. Instead, you could check whether current step == last item in steps (steps[-1]) and then make the next action 'complete', else make the step object equal to steps.index(current_step) + 2.

File: scripts/update_freezer_location.py (hunk 4 above)
Comment (EGCGDanForth) [reply scripts/update_freezer_location.py]: Yes, they will all have their current freezer location removed and will then state "Sample Destroyed" or something to this effect.

Comment (EGCGDanForth) [reply scripts/update_freezer_location.py]: Script renamed to "remove_from_freezer.py"

File: scripts/next_step_assignment.py (hunk 2 above)
Comment (EGCGDanForth) [reply scripts/next_step_assignment.py]: Updated

File: scripts/next_step_assignment.py (hunk 1 above)
Comment (EGCGDanForth) [reply scripts/next_step_assignment.py]: Removed

File: scripts/next_step_assignment.py (hunk 3 above)
Comment (EGCGDanForth) [reply scripts/next_step_assignment.py]: Removed

**Comment by EGCGDanForth** [comment]
All comments from tcezard, mwhamgenomics and kemelianova have been addressed.

**Comment by tcezard** [comment]
@mwhamgenomics and @katieemelianova  Could you review ?

**Pull request merged**